### DISCUSSION TITLE
Can i use one view in a query _find? and can i reduce by average in a view?

### DISCUSSION THREAD

**Discussion by hamiltonsalazar** [apache/couchdb#2990] [Q&A] [answered]
I have an array field  `scores` where it has integers. I need to get the documents with a average greater than a "X" value, but the lenght of array field can vary. 

Example:
```
{
  "_id": "101",
  "_rev": "1-903e21561b2ce58f68887ffb59a6ff24",
  "name": "Matías",
...
  "scores": [
    5,
    7,
    2,
    7,
    5,
    1,
    4,
    6
  ],
...
"texts": [ ... ]
}
```

I want to use those returned documents and find by the `text` field using the `$regex` operator with _find. 

Or if there is another way for me to consult it.

**Comment by rjharmon** [comment]
It looks like your question isn't really scoped at a level that's a close match for this forum.  I'll suggest a general direction you can explore further to design your system to meet your higher-level objective.

I think this can be another case for map/reduce, with `_id` and your integer scores emitted instead of the `1` that sometimes is used. The _stats reducer for map/reduce can give you sum and count with grouping, from which the average for each _id is easily computed.

**Comment by janl** [comment, marked as answer]
You can’t do this in Mango and will have to try in JS views.

**Reply by hamiltonsalazar** [reply]
Can I use the operator `$regex` in JS views or something similar that works same?

**Reply by janl** [reply]
you can write full JS there, which includes regular expressions
